shell/platform/linux/public/flutter_linux/fl_channel.h

```diff
@@ -0,0 +1,114 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_SHELL_PLATFORM_LINUX_FL_CHANNEL_H_
+#define FLUTTER_SHELL_PLATFORM_LINUX_FL_CHANNEL_H_
+
+#if !defined(__FLUTTER_LINUX_INSIDE__) && !defined(FLUTTER_LINUX_COMPILATION)
+#error "Only <flutter_linux/flutter_linux.h> can be included directly."
+#endif
+
+#include <glib-object.h>
+
+#include "fl_binary_messenger.h"
+
+G_BEGIN_DECLS
+
+G_DECLARE_DERIVABLE_TYPE(FlChannel, fl_channel, FL, CHANNEL, GObject)
+
+/**
+ * FlChannel:
```

Comment:
What is the purpose of this object? We don't have it on any of the other platforms; `BasicMessageChannel` and `MethodChannel` have similar structures, but the APIs are at different levels of abstractions so they don't share any common interface. A `MethodChannel` shouldn't have APIs in terms of `GBytes`, as all of these do.

Reply:
I initially didn't have a common `FlChannel` class and I thought it odd that the Dart code doesn't have one either. It seems strange that the two channel types aren't connected. The reason I updated this branch was there's some common code in `FlBasicMessageChannel` and `FlMethodChannel` regarding storing the name and setting up the handlers with the `FlBinaryMessenger`. These classes weren't derivable originally, but as I'm implementing the system channels it made sense to me that those should override channel classes, and having a common `FlChannel` class makes them easier to manage together.

The `FlChannel` methods can all be made private, which is something I was considering at the time. So that would resolve the `GBytes`/`FlValue` issue.

Reply:
After further discussion on Discord it is decided to remove `FlChannel` for consistency with the existing code and a lack of value in having it.